// @aws-sdk/client-rds/dist-es/commands/CopyDBSnapshotCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getCrossRegionPresignedUrlPlugin } from "@aws-sdk/middleware-sdk-rds";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CopyDBSnapshotMessageFilterSensitiveLog, CopyDBSnapshotResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCopyDBSnapshotCommand, serializeAws_queryCopyDBSnapshotCommand, } from "../protocols/Aws_query";
export class CopyDBSnapshotCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, CopyDBSnapshotCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getCrossRegionPresignedUrlPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RDSClient";
        const commandName = "CopyDBSnapshotCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: CopyDBSnapshotMessageFilterSensitiveLog,
            outputFilterSensitiveLog: CopyDBSnapshotResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryCopyDBSnapshotCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryCopyDBSnapshotCommand(output, context);
    }
}
