// @aws-sdk/client-rds/dist-es/commands/ModifyCertificatesCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyCertificatesMessageFilterSensitiveLog, ModifyCertificatesResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryModifyCertificatesCommand, serializeAws_queryModifyCertificatesCommand, } from "../protocols/Aws_query";
export class ModifyCertificatesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ModifyCertificatesCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RDSClient";
        const commandName = "ModifyCertificatesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ModifyCertificatesMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifyCertificatesResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryModifyCertificatesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryModifyCertificatesCommand(output, context);
    }
}
