// @aws-sdk/client-rds/dist-es/commands/StartDBInstanceAutomatedBackupsReplicationCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getCrossRegionPresignedUrlPlugin } from "@aws-sdk/middleware-sdk-rds";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartDBInstanceAutomatedBackupsReplicationMessageFilterSensitiveLog, StartDBInstanceAutomatedBackupsReplicationResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryStartDBInstanceAutomatedBackupsReplicationCommand, serializeAws_queryStartDBInstanceAutomatedBackupsReplicationCommand, } from "../protocols/Aws_query";
export class StartDBInstanceAutomatedBackupsReplicationCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, StartDBInstanceAutomatedBackupsReplicationCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getCrossRegionPresignedUrlPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RDSClient";
        const commandName = "StartDBInstanceAutomatedBackupsReplicationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StartDBInstanceAutomatedBackupsReplicationMessageFilterSensitiveLog,
            outputFilterSensitiveLog: StartDBInstanceAutomatedBackupsReplicationResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_queryStartDBInstanceAutomatedBackupsReplicationCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_queryStartDBInstanceAutomatedBackupsReplicationCommand(output, context);
    }
}
